rfc/helm-synthesis.md

```diff
@@ -0,0 +1,234 @@
+
+### What alternative solutions did you consider?
+
+* **`helm create` command to generate chart structure**
+
+  [helm create](https://helm.sh/docs/helm/helm_create/) is a helm command that generates a templated chart with the necessary folder structure and files needed to deploy the chart with helm. I did not use this to create the chart structure for two reasons,
+  1. This would mean that we need to take a dependency with helm or mention to the user that helm needs to be present on their machine for the synthesis to work in this scenario.
+  2. This command creates all the necessary structure but also creates files for simple `nginx` application. For our purposes, these files would either needed to be deleted or modified to work with the manifests we are generating. This feels like an over kill to accomplish our requirements.
+
+* **[helm-x](https://github.com/mumoshu/helm-x) and [chartify](https://github.com/helmfile/chartify) libraries**
+  
+  * One of the functionality that these libraries provide users is to convert Kubernetes resource YAMLs to Helm Charts. These can be used to generate helm charts from our generated manifests from cdk8s app. Currently, these are version `0.x.x`, which means adding these to support our feature could result in added maintenance load for the team. I believe adding a solution of our own would give us more control and not limit us to a library's functionality.
+  * Another reason to not use these is that it would add a dependency on a binary file.
+
+### What are the drawbacks of this solution?
+
+* CDK8s generated manifests are static
+
+    When cdk8s app generates manifests, the resource names remain the same with each synthesis. In helm, the resource names are prefixed with the release name. This helps deploying the same helm chart multiple times in the same namespace. This is not possible with cdk8s currently. 
+
+* No customizability for generated helm chart
+
+    In this solution, there is no way for us to maintain any information that user adds to the generated helm charts. We would be overwriting any changes to the chart if the synthesis is performed again. This is true for our current manifest synthesis too and is an established contract with the user. 
+
+### What is the high-level project plan?
+
+* **Support helm format**
+
+   - Add support for 
+        - `--format` flag and set default to cdk8s. 
+        - `--chart-api-version` flag and set default to `v2`.
+        - `--chart-version` flag and set as required when `--format` is set to helm. No defaults for this flag.
+        - Set flags to [hide](https://github.com/yargs/yargs/pull/190) for now.
+   - Create helm chart structure and template relevant files 
+   - Download CRDs present in the cdk8s.yaml file
+   - Synthesize application to helm chart structure
+   - Un-hide the flag
+
+### Are there any open issues that need to be addressed later?
+
+* **Generated Manifest Diff**
```

Comment:
What is the idea? are you thinking about a `cdk8s diff` command that would compare two directories on a file system? Not sure I understand why this is a concern of cdk8s. As you mention, diff can (and should) be done via normal version control. Note that the helm diff plugin you mention compares between a local chart, and the one **deployed** on the cluster, which cdk8s is not designed to do anyway.

Reply:
Ahh, I see. Yeah, I just wanted a way where user can know what changes are being introduced as a new synthesis command is run. Which can be done via version control.